app/scripts/views/components/survivalChart/template.js

```diff
@@ -122,21 +94,141 @@
         var _opts = $.extend(true, {}, opts);
         _opts.groups = this.calcCurvesData(
           _opts.groups, _opts.groupType);
-
+        this.groups = $.extend(true, [], _opts.groups);
         if (_opts.subtitle) {
           this.displayName = this.attributes.display_name + _opts.subtitle;
         }
-        this.chartInst.update(
-          _opts.groups, this.chartId, this.attributes.attr_id);
-        this.checkDownloadableStatus();
-        this.updateQtipContent();
+        this.showingRainbowSurvival = true;
+        this.updateRainbowSurvival();
       }
     },
     methods: {
+      updateRainbowSurvival: function() {
+        var groups = $.extend(true, [], this.groups);
+        if (this.excludeNa) {
+          groups = _.filter(groups, function(group) {
+            return group.name !== 'NA';
+          });
+        }
+        this.updatePlot(groups);
+      },
+      updatePlot: function(groups) {
+        // Display name may be changed due to the rainbow survival
+        this.displayName = this.attributes.display_name;
+
+        this.chartInst.update(
+          groups ? groups : this.groups, this.chartId, this.attributes.attr_id);
+        this.checkDownloadableStatus();
+        this.showLoad = false;
+        this.updateQtipContent();
+      },
+      updatePlotGroups: function(hasFilters) {
```

Comment:
Since this function going to mutate the groups data, better to move this to a parent object i.e may be to vueCore.js or main.js